### PR TITLE
qemu_storage: adapt the collection mapping function

### PR DESCRIPTION
**Pull request by hellohellenmao**
Adapt the function according to different python versions

ID: 2413

### CODE REVIEW

**Comment by hellohellenmao** [comment]
(1/1) Host_RHEL.m10.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_backing_file.q35: STARTED
 (1/1) Host_RHEL.m10.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_backing_file.q35: PASS (257.23 s)

**Comment by hellohellenmao** [comment]
@zhencliu Could you please help to review here? Thanks.

File: virttest/qemu_storage.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import os
 import re
 import string
+from distutils import sysconfig
```
Comment (zhencliu) [inline virttest/qemu_storage.py:14]: what about using try except, e.g. 
```
try:
    from collections.abc import Mapping
except ImportError:
    from collections import Mapping
```
Check BytesIO in utils_misc.py for an example

BTW, we cannot compare string for the version, e.g.
```
>>> '3.6' < '3.10'
False
```

But we can use tuple comparison, e.g. 
```
sys.version_info < (3, 10)

>>> (3, 6) < (3, 10)
True
```

Comment (hellohellenmao) [reply virttest/qemu_storage.py]: Yes, agree with you, your method makes more sense. I have updated the fix. Thanks.

**Comment by hellohellenmao** [comment]
(1/1) Host_RHEL.m10.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_backing_file.q35: PASS (264.22 s)

@zhencliu Could you please help to take another review? Thanks.

**Review by zhencliu — APPROVED** [review]

**Comment by hellohellenmao** [comment]
@YongxueHong Could you please help to review here? Thanks.

File: virttest/qemu_storage.py [hunk 2]
```diff
@@ -13,6 +13,12 @@
 import string
 
 import six
+
+try:
+    from collections.abc import Mapping
+except ImportError:
+    from collections import Mapping
```
Comment (YongxueHong) [inline virttest/qemu_storage.py:20]: Seems like https://github.com/avocado-framework/avocado-vt/blob/48b5b3dde0a4506031588f6685db67365fd33839/virttest/qemu_storage.py#L8
is useless, we could delete it.

Comment (zhencliu) [reply virttest/qemu_storage.py:19]: Hi @hellohellenmao and @YongxueHong , actually, we use collections.OrderedDict when getting image opts

Comment (hellohellenmao) [reply virttest/qemu_storage.py:19]: Updated, thanks!

**Comment by hellohellenmao** [comment]
@YongxueHong Please help to review again, thanks.

File: virttest/qemu_storage.py (hunk 2 above)
Comment (YongxueHong) [reply virttest/qemu_storage.py:20]: > Hi @hellohellenmao and @YongxueHong , actually, we use collections.OrderedDict when getting image opts

Thanks, I got it, I missed it.

**Review by YongxueHong — APPROVED** [review]

**Pull request merged**